DECISIONS.md

```diff
@@ -0,0 +1,119 @@
+# Decisions
+
+This document is intended to provide clarity on many of the decisions/rationalizations
+which exist inside of [Datamill's](https://datamill.co/) Target SQL project
+for [Singer](https://singer.io).
+
+The guiding principles we try to adhere to herein as far as _how_ to reach a
+conclusion are:
+
+1. When possible, make the resulting data/schema in the remote target consistent, no matter the ordering of potential messages
+  - ie, if our decision would result in a random schema being produced in the remote target for no reasonable benefit, this is in violation
+1. Do right by the common _majority_ of users
+1. Make a best effort to prevent a user from having to intervene
+1. Use [Stitch’s offering and documentation](https://www.stitchdata.com/docs) as best practice guidance
+
+## Schema
+
+### De-nesting
+
+#### What
+
+- [JSON Schema](https://json-schema.org/) allows for complex schemas which have non-literal (ie, compositional) elements
+- Standard SQL does not support compositional elements, but rather demands something closer to full denormalization
+- To overcome this, `target-sql` provides tooling which unpacks json `objects` into their parent record, and json `arrays` as sub tables
+
+#### Why
+
+This is the same approach that Stitch Data takes with `array` de-nesting.
+
+### Column Type Mismatch
+
+#### What
+
+- A field has been streamed to the remote target with type `integer`
+- A new field with the _same raw name_ as the remote column has been streamed but has type `BAR`
+- Data of type `BAR` cannot be placed into a column of type `boolean`
+- `target-sql` has tooling which will:
+  - rename the original column to `original_field_name__i`
+  - make the renamed column `nullable`
+  - create a new column of name `original_field_name__b`
+  - stream new data to `original_field_name__b`
+  - (to see a full list of type suffixes, please see: [`json_schema._shorthand_mappings`](https://github.com/datamill-co/target-postgres/blob/master/target_postgres/json_schema.py#L283))
```

Comment:
May want to use the most recent commit hash in the URL, otherwise the line number will shift over time.